File: src/test/java/no/novari/flyt/archive/gateway/dispatch/mapping/KontaktinformasjonMappingServiceTest.java
package no.novari.flyt.archive.gateway.dispatch.mapping;

import no.fint.model.felles.kompleksedatatyper.Kontaktinformasjon;
import no.novari.flyt.archive.gateway.dispatch.model.instance.KontaktinformasjonDto;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class KontaktinformasjonMappingServiceTest {

    private KontaktinformasjonMappingService kontaktinformasjonMappingService;

    @BeforeEach
    public void setup() {
        kontaktinformasjonMappingService = new KontaktinformasjonMappingService();
    }

    @Test
    public void testToKontaktinformasjonWithNullInput() {
        Kontaktinformasjon kontaktinformasjon = kontaktinformasjonMappingService.toKontaktinformasjon(null);

        assertThat(kontaktinformasjon).isNull();
    }

    @Test
    public void testToKontaktinformasjonWithValidInput() {
        KontaktinformasjonDto dto = KontaktinformasjonDto.builder()
                .epostadresse("testEmail@test.com")
                .telefonnummer("123456789")
                .mobiltelefonnummer("987654321")
                .build();

        Kontaktinformasjon actualKontaktinformasjon = kontaktinformasjonMappingService.toKontaktinformasjon(dto);

        assertThat(dto.getEpostadresse()).isPresent();
        assertThat(dto.getEpostadresse().get()).isEqualTo(actualKontaktinformasjon.getEpostadresse());
        assertThat(dto.getTelefonnummer()).isPresent();
        assertThat(dto.getTelefonnummer().get()).isEqualTo(actualKontaktinformasjon.getTelefonnummer());
        assertThat(dto.getMobiltelefonnummer()).isPresent();
        assertThat(dto.getMobiltelefonnummer().get()).isEqualTo(actualKontaktinformasjon.getMobiltelefonnummer());
    }
}

File: src/test/kotlin/no/novari/flyt/archive/gateway/dispatch/mapping/KontaktinformasjonMappingServiceTest.kt
package no.novari.flyt.archive.gateway.dispatch.mapping

import no.novari.flyt.archive.gateway.dispatch.model.instance.KontaktinformasjonDto
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class KontaktinformasjonMappingServiceTest {
    private lateinit var kontaktinformasjonMappingService: KontaktinformasjonMappingService

    @BeforeEach
    fun setup() {
        kontaktinformasjonMappingService = KontaktinformasjonMappingService()
    }

    @Test
    fun `given a null input, returns null`() {
        assertThat(kontaktinformasjonMappingService.toKontaktinformasjon(null)).isNull()
    }

    @Test
    fun `given a valid input, maps to Kontaktinformasjon`() {
        val dto =
            KontaktinformasjonDto
                .builder()
                .epostadresse("testEmail@test.com")
                .telefonnummer("123456789")
                .mobiltelefonnummer("987654321")
                .build()

        val actualKontaktinformasjon = kontaktinformasjonMappingService.toKontaktinformasjon(dto)

        assertThat(actualKontaktinformasjon!!.epostadresse).isEqualTo("testEmail@test.com")
        assertThat(actualKontaktinformasjon.telefonnummer).isEqualTo("123456789")
        assertThat(actualKontaktinformasjon.mobiltelefonnummer).isEqualTo("987654321")
    }
}
